src/pythonic/index.md

```diff
@@ -174,6 +174,9 @@ class Sieve:
         primal.sieve_destroy(self.sieve)
 
     def is_prime(self, n):
+        if n > self.upper_bound():
+            raise IndexError("{} not in upper bound {}"
```

Comment:
Should we mention that you need to add this check to prevent panics in the main body of the chapter?